enterprise/views.py

```diff
@@ -375,32 +390,122 @@ def get(self, request):
+                consent_record = get_data_sharing_consent(
+                    request.user.username,
+                    enterprise_customer_uuid,
+                    program_uuid=program_uuid,
+                    course_id=course_id
+                )
+            except NotConnectedToOpenEdX as error:
+                error_code = 'ENTGDS001'
+                log_message = (
+                    'The was a problem with getting the consent record of user {userid} with '
+                    'uuid {enterprise_customer_uuid}. get_data_sharing_consent threw '
+                    'the following NotConnectedToOpenEdX error: {error}'
+                    'for course_id {course_id}.'
+                    'Error code {error_code} presented to user'.format(
+                        userid=request.user.id,
+                        enterprise_customer_uuid=enterprise_customer_uuid,
+                        error=error,
+                        error_code=error_code,
+                        course_id=course_id,
+                    )
+                )
+                return render_page_with_error_code_message(request, context_data, error_code, log_message)
 
-            enterprise_customer = consent_record.enterprise_customer
-        else:
-            if not request.user.is_staff:
-                raise PermissionDenied()
-            enterprise_customer = get_enterprise_customer_or_404(enterprise_customer_uuid)
+            try:
+                consent_required = consent_record.consent_required()
+            except AttributeError:
+                consent_required = None
+
+            if consent_record is None or not consent_required:
+                error_code = 'ENTGDS002'
+                log_message = (
+                    'The was a problem with the consent record of user {userid} with '
+                    'enterprise_customer_uuid {enterprise_customer_uuid}. consent_record has a value '
+                    'of {consent_record} and consent_record.consent_required() a '
+                    'value of {consent_required} for course_id {course_id}. '
+                    'Error code {error_code} presented to user'.format(
+                        userid=request.user.id,
+                        enterprise_customer_uuid=enterprise_customer_uuid,
+                        consent_record=consent_record,
+                        consent_required=consent_required,
+                        error_code=error_code,
+                        course_id=course_id,
+                    )
+                )
+                return render_page_with_error_code_message(request, context_data, error_code, log_message)
+            else:
+                enterprise_customer = consent_record.enterprise_customer
+        elif not request.user.is_staff:
+            raise PermissionDenied()
 
+        # Retrieve context data again now that enterprise_customer logic has been run
         context_data = get_global_context(request, enterprise_customer)
```

Comment:
it's weird to me that we have to call this again, i doubt we would end up with a different enterprise customer object from the consent object since both lookups are by the uuid.